src/utils/featheryClient/integrationClient.ts

```diff
@@ -31,33 +57,16 @@ export default class IntegrationClient {
     this.draft = draft;
     this.bypassCDN = bypassCDN;
     this.submitQueue = Promise.resolve();
+    this.offlineRequestHandler = new OfflineRequestHandler(formKey);
   }
 
-  async _checkResponseSuccess(response: any) {
-    let payload;
-    switch (response.status) {
-      case 200:
-      case 201:
-        return;
-      case 400:
-        payload = JSON.stringify(await response.clone().text());
-        console.error(payload.toString());
-        return;
-      case 401:
-        throw new errors.SDKKeyError();
-      case 404:
-        throw new errors.FetchError("Can't find object");
-      case 409:
-        location.reload();
-        return;
-      case 500:
-        throw new errors.FetchError('Internal server error');
-      default:
-        throw new errors.FetchError('Unknown error');
-    }
-  }
-
-  _fetch(url: any, options: any, parseResponse = true) {
+  _fetch(
+    url: any,
+    options: any,
+    parseResponse = true,
+    type?: string,
```

Comment:
maybe type and stepKey can be merged into a single offlineArgs array parameter? and then it can be passed in as ...offlineArgs